### PR TITLE
Add the `—bin-path` option to `swift build`

### PR DESCRIPTION
**Pull request by hartbit**
Resolves [SR-5259](https://bugs.swift.org/browse/SR-5259)

### CODE REVIEW

File: Tests/CommandsTests/BuildToolTests.swift [hunk 1]
```diff
@@ -27,8 +27,16 @@ final class BuildToolTests: XCTestCase {
         XCTAssert(try execute(["--version"]).contains("Swift Package Manager"))
     }
 
+    func testBinPath() throws {
+        fixture(name: "ValidLayouts/SingleModule/Executable") { path in
+            XCTAssert(try execute(["--bin-path"], packagePath: path).contains("/.build/debug"))
+            XCTAssert(try execute(["-c", "release", "--bin-path"], packagePath: path).contains("/.build/release"))
```
Comment (aciidb0mb3r) [inline Tests/CommandsTests/BuildToolTests.swift]: I think we should check for the exact path here just to be safe. We can construct that using the `path` variable.

**Comment by aciidb0mb3r** [comment]
LGTM, one minor request.

Thanks!

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Pull request merged**